private/pkg/rpc/rpcauth/rpcauth.go

```diff
@@ -16,106 +16,23 @@ package rpcauth
 
 import (
 	"context"
-	"errors"
-	"net/http"
-	"strings"
 
 	"github.com/bufbuild/buf/private/pkg/rpc"
 )
 
-const (
-	// authenticationHeader is the standard OAuth header used for authenticating
-	// a user. Ignore the misnomer.
-	authenticationHeader = "Authorization"
-	// authenticationTokenPrefix is the standard OAuth token prefix.
-	// We use it for familiarity.
-	authenticationTokenPrefix = "Bearer "
-)
-
-type authContextKey struct{}
-
-// User describes an authenticated user.
-type User struct {
-	Subject string
-}
-
-// GetUser gets the currently authenticated user, if
-// there is one. It returns nil, false, if there is no
-// authenticated user.
-func GetUser(ctx context.Context) (*User, bool) {
-	userValue := ctx.Value(authContextKey{})
-	if userValue == nil {
-		return nil, false
-	}
-	// This is the only package where we can set this context key, so
-	// this is guaranteed to be of this type if it exists.
-	user, ok := userValue.(*User)
-	return user, ok
-}
-
-// WithToken adds the token to the context via a header.
+// WithToken adds the token to the outgoing context via a header.
 func WithToken(ctx context.Context, token string) context.Context {
 	if token != "" {
-		return rpc.WithOutgoingHeader(ctx, authenticationHeader, authenticationTokenPrefix+token)
+		return rpc.WithOutgoingContextHeader(ctx, rpc.AuthenticationHeader, rpc.AuthenticationTokenPrefix+token)
```

Comment:
The `WithToken` and `WithTokenIfNoneSet` functions will be replaced by Connect interceptors in a future PR.  Leaving here now because the ethos of this PR is to not change any functionality in the CLI.